cirq-core/cirq/contrib/acquaintance/executor.py

```diff
@@ -100,6 +117,67 @@ def optimization_at(
         )
 
 
+class StrategyExecutorTransformer:
+    """Executes an acquaintance strategy."""
+
+    def __init__(self, execution_strategy: ExecutionStrategy) -> None:
+        """Initializes transformer.
+
+        Raises:
+            ValueError: if execution_strategy is None.
+        """
+
+        if execution_strategy is None:
+            raise ValueError('execution_strategy cannot be None')
+        self.execution_strategy = execution_strategy
+        self.mapping = execution_strategy.initial_mapping.copy()
+
+    def __call__(
+        self, circuit: circuits.Circuit, context: Optional['cirq.TransformerContext'] = None
+    ) -> circuits.Circuit:
+        """Executes an acquaintance strategy using primitive map function and
+          mutates initial mapping.
+
+        Args:
+            circuit: 'circuits.Circuit' input circuit to transform.
+            context: `cirq.TransformerContext` storing common configurable
+              options for transformers.
+
+        Returns:
+            A copy of the modified circuit after executing an acquaintance
+              strategy on all instances of AcquaintanceOpportunityGate
+        """
+
+        expose_acquaintance_gates(circuit)
+        return transformers.map_operations_and_unroll(
+            circuit=circuit,
+            map_func=self.map_func,
+            deep=context.deep if context else False,
+            tags_to_ignore=context.tags_to_ignore if context else (),
+        )
+
+    def get_mapping(self) -> LogicalMapping:
+        return self.mapping.copy()
```

Comment:
Do we need to return a copy here? 
Also, let's change this to a property `mapping` and rename `self.mapping` to `self._mapping`. 

```suggestion

    @property
    def mapping(self) -> LogicalMapping:
        return self._mapping
```

Reply:
I don't think so.

Reply:
Can you add a `@property` decorator as well to make it a property instead of method?